src/hivpy/hiv_status.py

```diff
@@ -24,17 +40,87 @@ def initial_HIV_status(self, population: pd.DataFrame):
+            for age_group in range(5):   # FIXME need to get number of age groups from somewhere
+                sub_pop = population.data.loc[(population.data[col.SEX] == sex) & (
+                    population.data[col.SEX_MIX_AGE_GROUP] == age_group)]
+                # total number of people partnered to people in this group
+                n_stp_total = sum(sub_pop[col.NUM_PARTNERS])
+                # num people partered to HIV+ people in this group
+                n_stp_of_infected = sum(sub_pop.loc[sub_pop[col.HIV_STATUS], col.NUM_PARTNERS])
+                # Probability of being HIV prositive
+                if n_stp_of_infected == 0:
+                    self.stp_HIV_rate[sex][age_group] = 0
+                else:
+                    self.stp_HIV_rate[sex][age_group] = n_stp_of_infected / \
+                        n_stp_total  # TODO: need to double check this definition
+                # Chances of being in a given viral group
+                if n_stp_total > 0:
+                    self.stp_viral_group_rate[sex][age_group] = [
+                        sum(sub_pop.loc[sub_pop[col.VIRAL_LOAD_GROUP] == vg,
+                            col.NUM_PARTNERS])/n_stp_total for vg in range(7)]
+                else:
+                    self.stp_viral_group_rate[sex][age_group] = np.array([1, 0, 0, 0, 0, 0, 0])
+
+    def set_dummy_viral_load(self, population):
+        """Dummy function to set viral load until this
+        part of the code has been implemented properly"""
+        population.data[col.VIRAL_LOAD_GROUP] = rng.choice(7, population.size)
+
+    def get_infection_prob(self, sex, age, n_partners, stp_age_groups):
+        # Slow example that avoid repeating the iterations over partners
+        # three time by putting them as part of
+        # one for loop, but for loops in python will be slow.
+        target_sex = opposite_sex(sex)
+        infection_prob = np.zeros(n_partners)
+        for i in range(n_partners):
+            stp_viral_group = rng.choice(
+                7, p=self.stp_viral_group_rate[target_sex][stp_age_groups[i]])
+            HIV_probability = self.stp_HIV_rate[opposite_sex(target_sex)][stp_age_groups[i]]
+            infection_prob[i] = HIV_probability * max(0, rng.normal(
+                self.transmission_means[stp_viral_group],
+                self.transmission_sigmas[stp_viral_group]))
+            if (sex == SexType.Female):
+                if (age < 20):
+                    infection_prob[i] *= self.fold_change_yw
+                else:
+                    infection_prob[i] *= self.fold_change_w
+            return infection_prob
+
+    def stp_HIV_transmission(self, person):
+        # TODO: Add circumcision, STIs etc.
+        """Returns True if HIV transmission occurs, and False otherwise"""
+        stp_viral_groups = np.array([rng.choice(7, p=self.stp_viral_group_rate[opposite_sex(
+            person[col.SEX])][age_group]) for age_group in person[col.STP_AGE_GROUPS]])
```

Comment:
I find something like the following a more readable way to break it up:
```suggestion
        stp_viral_groups = np.array([
            rng.choice(7, p=self.stp_viral_group_rate[opposite_sex(person[col.SEX])][age_group])
            for age_group in person[col.STP_AGE_GROUPS]
        ])
```
but that's personal preference! (and the second line is actually longer this way... though less than 100)